### PR TITLE
Basforza codereview

### PR DESCRIPTION
**Pull request by zbasford**
Three Technical Suggestions:
1.) Your classes are well designed, but relating to layout and design of your application you should reduce the number of screens to where the user has one selection instead of three buttons for three separate layouts to reduce complexity. 
2.) Some of your methods did not throw exceptions which would have made your application stop working so make sure that you are adding exceptions to your methods.
3.) You could add more to your javadoc in order for others that use your application or look at your code to understand what is going on throughout the process. Also using more appropriate class names and variables would help in this process as well. 
4.) I would add a popup menu in place of the one that you currently have to make the design look better for the type of app you are developing.

Part Three: Submit your Analysis
Overall I was pleased with your application the UI is pretty simple which is essential in mobile app development to be an easy experience for the user. The code was up in a fair amount of time to be able to complete the code review by Sunday evening. The comments in the code were enough for me, but could be a little more descriptive in order for it to be easier to understand. The rationale behind the changes that I had made was to make it easier to understand what each class is doing with better classnames, variable names, along with other minor code changes.

Three Technical concepts that I have learned:
1.) I learned how to incorporate a floating action button, which is something that could be useful down the road.
2.) Learned about a new layout, which is the coordinator layout and seems to be very easy to use looking at the UI of your application.
3.) Gained more knowledge of how to use a snackbar to display a message in the app.

### CODE REVIEW

File: app/src/main/java/ibrahioy/uc/edu/BookTemplateDesign1.java [hunk 1]
```diff
@@ -3,7 +3,7 @@
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
 
-public class Book1Activity extends AppCompatActivity {
+public class BookTemplateDesign1 extends AppCompatActivity {
```
Comment (zbasford) [inline app/src/main/java/ibrahioy/uc/edu/BookTemplateDesign1.java]: Changed the Name to better understand what the class is dealing with and same for the other two class names.

File: app/src/main/java/ibrahioy/uc/edu/MainActivity.java [hunk 2]
```diff
@@ -23,7 +23,7 @@ protected void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.activity_main);
         Toolbar toolbar = findViewById(R.id.toolbar);
         setSupportActionBar(toolbar);
-
+// sets the onclick methods for the buttons so if clicked then opens the corresponding activity
```
Comment (zbasford) [inline app/src/main/java/ibrahioy/uc/edu/MainActivity.java]: javadocumentation saying what the button onclick method is doing.

File: app/src/main/java/ibrahioy/uc/edu/MainActivity.java [hunk 3]
```diff
@@ -71,20 +71,20 @@ public void onClick(View view) {
 
     private void openBook3Activity() {
 
-        Intent intent = new Intent(this, Book1Activity.class);
+        Intent intent = new Intent(this, BookTemplateDesign3.class);
```
Comment (zbasford) [inline app/src/main/java/ibrahioy/uc/edu/MainActivity.java]: rechanged the names

File: app/src/main/java/ibrahioy/uc/edu/dao/IBookInfoDAO.java [hunk 4]
```diff
@@ -3,16 +3,16 @@
 import java.util.List;
 import ibrahioy.uc.edu.dto.BookDTO;
 
-public interface IBookDAO {
+public interface IBookInfoDAO {
 
     /* Provided a book title, search for matching book results and return as a list */
-    public List<BookDTO> findByTitle(String searchTitle);
+    public List<BookDTO> findByTitle(String searchTitle) throws Exception;
```
Comment (zbasford) [inline app/src/main/java/ibrahioy/uc/edu/dao/IBookInfoDAO.java]: add exception handling

File: app/src/main/java/ibrahioy/uc/edu/dto/BookDTO.java [hunk 5]
```diff
@@ -9,7 +9,7 @@
 
     private String title;
     private String author;
-    private Boolean isPublic; //Whether or not general community can view book
+    private Boolean isViewable; //Whether or not general community can view book
```
Comment (zbasford) [inline app/src/main/java/ibrahioy/uc/edu/dto/BookDTO.java]: rename variable

File: app/src/main/java/ibrahioy/uc/edu/dto/BookDTO.java [hunk 6]
```diff
@@ -32,11 +32,11 @@ public void setAuthor(String author) {
     }
 
     public Boolean getPublic() {
-        return isPublic;
+        return isViewable;
     }
 
     public void setPublic(Boolean aPublic) {
-        isPublic = aPublic;
+        isViewable = aPublic;
```
Comment (zbasford) [inline app/src/main/java/ibrahioy/uc/edu/dto/BookDTO.java]: fix variable

File: app/src/main/res/layout/activity_book3.xml [hunk 7]
```diff
@@ -6,6 +6,6 @@
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     android:background="#42F4BC"
-    tools:context=".Book3Activity">
+    tools:context=".BookTemplateDesign3">
```
Comment (zbasford) [inline app/src/main/res/layout/activity_book3.xml]: Changed class names in layout

File: app/src/main/res/layout/activity_book2.xml [hunk 8]
```diff
@@ -5,6 +5,6 @@
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     android:background="#CCB978"
-    tools:context=".Book2Activity">
+    tools:context=".BookTemplateDesign2">
```
Comment (zbasford) [inline app/src/main/res/layout/activity_book2.xml]: Changed layout to appropriate class name

File: app/src/main/res/layout/activity_book1.xml [hunk 9]
```diff
@@ -5,7 +5,7 @@
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     android:background="#CCB978"
-    tools:context=".Book1Activity">
+    tools:context=".BookTemplateDesign1">
```
Comment (zbasford) [inline app/src/main/res/layout/activity_book1.xml]: Changed name to appropriate class name

File: app/src/main/java/ibrahioy/uc/edu/BookTemplateDesign1.java (hunk 1 above)
Comment (discospiff) [reply app/src/main/java/ibrahioy/uc/edu/BookTemplateDesign1.java]: Additionally, I'd take the number out.  You won't know the difference between BookTemplateDesign1 and BookTemplateDesign2  in a couple of years.

File: app/src/main/java/ibrahioy/uc/edu/MainActivity.java (hunk 2 above)
Comment (discospiff) [reply app/src/main/java/ibrahioy/uc/edu/MainActivity.java]: Actually, that's inline doc.   JavaDoc belongs on top of methods and classes, and has the syntax:
/**
 */

JavaDoc is always a good idea.  Inline comments: the preference is to write readable code that does not require heavy inline documentation.  :)

File: app/src/main/java/ibrahioy/uc/edu/dto/BookDTO.java (hunk 6 above)
Comment (discospiff) [reply app/src/main/java/ibrahioy/uc/edu/dto/BookDTO.java]: Change the method name to match as well.  The getter/setter should match the attribute it's getting/setting.

File: app/src/main/java/ibrahioy/uc/edu/dao/IBookInfoDAO.java (hunk 4 above)
Comment (discospiff) [reply app/src/main/java/ibrahioy/uc/edu/dao/IBookInfoDAO.java]: A few considerations here:
1) This isn't really exception handling, it's just saying the method may throw an exception.
2) Does the method actually throw an exception?  If not, this is misleading.
3) Funny thing: this is java.io.Exception, which is a high level exception superclass.  Unchecked exceptions are automatically thrown, so you only need to declare checked exceptions that can be thrown.  java.io.Exception could be either checked or unchecked, so I'd go with a more specific subclass that represents a checked exception.

**Comment by discospiff** [comment]
Several good concepts here worth considering for a merge.  Read my inline comments for more detail.

**Pull request merged**